ksql-common/src/main/java/io/confluent/ksql/util/ReservedInternalTopics.java

```diff
@@ -0,0 +1,153 @@
+  }
+
+  /**
+   * Returns the internal KSQL configs topic (used for KSQL standalone)
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @return The configurations topic name.
+   */
+  public static String configsTopic(final KsqlConfig ksqlConfig) {
+    return toKsqlInternalTopic(ksqlConfig, KSQL_CONFIGS_TOPIC_SUFFIX);
+  }
+
+  /**
+   * Returns the KSQL processing log topic.
+   * <p/>
+   * This is not an internal topic in the sense that users are intentionally meant to read from
+   * this topic to identify deserialization and other processing errors, define a KSQL stream on
+   * it, and potentially issue queries to filter from it, etc. This is why it is not prefixed in
+   * the way KSQL internal topics are.
+   *
+   * @param config The Processing log config, which is used to extract the processing topic suffix
+   * @param ksqlConfig The KSQL config, which is used to extract the KSQL service id.
+   * @return The processing log topic name.
+   */
+  public static String processingLogTopic(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig
+  ) {
+    final String topicNameConfig = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    if (topicNameConfig.equals(ProcessingLogConfig.TOPIC_NAME_NOT_SET)) {
+      return String.format(
+          "%s%s",
+          ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+          ProcessingLogConfig.TOPIC_NAME_DEFAULT_SUFFIX
+      );
+    } else {
+      return topicNameConfig;
+    }
+  }
+
+  /**
+   * Compute a name for a KSQL internal topic.
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @param topicSuffix A suffix that is appended to the topic name.
+   * @return The computed topic name.
+   */
+  private static String toKsqlInternalTopic(final KsqlConfig ksqlConfig, final String topicSuffix) {
+    return String.format(
+        "%s%s_%s",
+        KSQL_INTERNAL_TOPIC_PREFIX,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+        topicSuffix
+    );
+  }
+
+  private final Pattern hiddenTopicsPattern;
+  private final Pattern readOnlyTopicsPattern;
+
+  public ReservedInternalTopics(final KsqlConfig ksqlConfig) {
+    final ProcessingLogConfig processingLogConfig =
+        new ProcessingLogConfig(ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
+
+    try {
+      this.hiddenTopicsPattern = Pattern.compile(
+          Streams.concat(
+              Stream.of(KSQL_INTERNAL_TOPIC_PREFIX + ".*"),
+              ksqlConfig.getList(KsqlConfig.KSQL_INTERNAL_HIDDEN_TOPICS_CONFIG).stream()
+          ).collect(Collectors.joining("|"))
+      );
+    } catch (final Exception e) {
+      final String message = "Invalid pattern list in '"
+          + KsqlConfig.KSQL_INTERNAL_HIDDEN_TOPICS_CONFIG + "'";
+
+      LOG.error(message + ": " + e.getMessage());
+      throw new KsqlException(message, e);
+    }
```

Comment:
Rather than having this error checking here, it would be cleaner to check for invalid regular expressions using a `Validator` on the config itself.  This would mean the value is checked when the `KsqlConfig` is created, i.e. higher up the stack, resulting in an earlier, more targeted error message. Then this error checking, and the one below, can be removed; resulting in much cleaner code.

We already have `ConfigValidators`. Consider adding a validator in their for checking a regular expression is valid, and one that allows you to validate each entry in the list, i.e. two separate ones.  Or, just one that checks each string in the list is a valid regular expression.